war/src/main/java/io/lumeer/engine/controller/UserGroupFacade.java

```diff
@@ -0,0 +1,298 @@
+    *
+    * @param organization
+    *       Id of the organization.
+    * @return The list of user names.
+    */
+   public List<String> getUsers(final String organization) {
+      DataDocument document = dataStorage.readDocumentIncludeAttrs(UserGroup.COLLECTION_NAME, organizationFilter(organization), Collections.singletonList(UserGroup.ATTR_USERS));
+      List<DataDocument> users = document.getArrayList(UserGroup.ATTR_USERS, DataDocument.class);
+      return users != null ? users.stream().map(u -> u.getString(UserGroup.ATTR_USERS_USER)).collect(Collectors.toList()) : Collections.emptyList();
+   }
+
+   /**
+    * Read users and their groups in organization.
+    *
+    * @param organization
+    *       Id of the organization.
+    * @return The map of user names and groups.
+    */
+   public Map<String, List<String>> getUsersAndGroups(final String organization) {
+      DataDocument document = dataStorage.readDocumentIncludeAttrs(UserGroup.COLLECTION_NAME, organizationFilter(organization), Collections.singletonList(UserGroup.ATTR_USERS));
+      List<DataDocument> users = document.getArrayList(UserGroup.ATTR_USERS, DataDocument.class);
+      return users != null ? users.stream().collect(Collectors.toMap(u -> u.getString(UserGroup.ATTR_USERS_USER), u -> u.getArrayList(UserGroup.ATTR_USERS_GROUPS, String.class))) : Collections.emptyMap();
+   }
+
+   /**
+    * Read groups of user in organization.
+    *
+    * @param organization
+    *       Id of the organization.
+    * @param user
+    *       User identificator.
+    * @return The list of group names.
+    */
+   public List<String> getGroupsOfUser(final String organization, final String user) {
+      DataDocument userDoc = readUser(organization, user);
+      return userDoc != null ? userDoc.getArrayList(UserGroup.ATTR_USERS_GROUPS, String.class) : null;
+   }
+
+   /**
+    * Read users for specific group.
+    *
+    * @param organization
+    *       Id of the organization.
+    * @param group
+    *       Group name.
+    * @return The list of user names.
+    */
+   public List<String> getUsersOfGroup(final String organization, final String group) {
+      List<DataDocument> users = dataStorage.aggregate(UserGroup.COLLECTION_NAME, dataStorageDialect.usersOfGroupAggregate(organization, group));
+      return users.stream().map(u -> u.getString(UserGroup.ATTR_USERS_USER)).collect(Collectors.toList());
+   }
+
+   /**
+    * Read groups in organization.
+    *
+    * @param organization
+    *       Id of the organization.
+    * @return The list of group names.
+    */
+   public List<String> getGroups(final String organization) {
+      DataDocument document = dataStorage.readDocumentIncludeAttrs(Group.COLLECTION_NAME, organizationFilterGroups(organization), Collections.singletonList(Group.ATTR_GROUPS));
+      List<String> groups = document.getArrayList(Group.ATTR_GROUPS, String.class);
+      return groups != null ? groups : Collections.emptyList();
```

Comment:
Just a question I have probably asked before. Why don't we return an empty list directly in `getArrayList` so you never need to check if it is not `null`? It would save us some lines of code.